### PR TITLE
SD-103813 | implementation of the SQS receive message

### PR DESCRIPTION
**Pull request by dogukanayd**

### CODE REVIEW

File: inssqs/inssqs.go [hunk 1]
```diff
@@ -117,6 +119,39 @@ func (c *Config) setDefaults() {
...
+
+	input := &awssqs.ReceiveMessageInput{
+		QueueUrl:            q.url,
+		MaxNumberOfMessages: maxMessages,
+		VisibilityTimeout:   visibilityTimeout,
+	}
+
+	res, err := q.client.ReceiveMessage(context.Background(), input)
+
+	if err != nil {
```
Comment (sezaakgun) [inline inssqs/inssqs.go:151]:
```suggestion
	res, err := q.client.ReceiveMessage(context.Background(), input)
	if err != nil {
```
in styling-wise, we prefer handling error directly with no new line between

File: inssqs/inssqs.go [hunk 2]
```diff
@@ -117,6 +119,39 @@ func (c *Config) setDefaults() {
...
+	}
+
+	if maxMessages < 1 {
+		return ReceiveMessageOutput{}, fmt.Errorf("maxMessages should be greater than 0")
+	}
+
+	input := &awssqs.ReceiveMessageInput{
+		QueueUrl:            q.url,
+		MaxNumberOfMessages: maxMessages,
+		VisibilityTimeout:   visibilityTimeout,
```
Comment (sezaakgun) [inline inssqs/inssqs.go]: isn't there a validation rule that we can apply to the `visibilityTimeout` like we did to `maxMessages` ?

File: inssqs/README.md [hunk 3]
```diff
@@ -61,6 +61,9 @@ if err != nil {
 }
 ```
 
+### Receiving Messages
+Receive batch of messages from SQS queue using `ReceiveMessageBatch()`
+
```
Comment (sezaakgun) [inline inssqs/README.md:66]: code example below would be nice like examples above

File: inssqs/inssqs.go [hunk 4]
```diff
@@ -117,6 +119,42 @@ func (c *Config) setDefaults() {
...
+//
+// Parameters:
+// - rmi: ReceiveMessageInput containing the parameters for receiving messages from the SQS queue.
+//
+// Returns:
+// - res: A pointer to ReceiveMessageOutput containing the received messages.
+// - err: An error indicating any failure during the receiving process, nil if all messages were received successfully.
+func (q *queue) ReceiveMessageBatch(rmi ReceiveMessageInput) (ReceiveMessageOutput, error) {
+	if rmi.MaxNumberOfMessages > 10 {
+		return ReceiveMessageOutput{}, fmt.Errorf("maxMessages should be less than or equal to 10")
```
Comment (rafet) [inline inssqs/inssqs.go:132]: fmt.Errorf is redundant here, formating isn't used. we can use `errors.New` in inssqs/errors.go

File: inssqs/inssqs.go [hunk 5]
```diff
@@ -117,6 +119,42 @@ func (c *Config) setDefaults() {
 	}
 }
 
+// ReceiveMessageBatch receives a batch of messages from an SQS queue
+//
+// Parameters:
+// - rmi: ReceiveMessageInput containing the parameters for receiving messages from the SQS queue.
+//
+// Returns:
+// - res: A pointer to ReceiveMessageOutput containing the received messages.
```
Comment (rafet) [inline inssqs/inssqs.go:128]: doesn't return the pointer

File: inssqs/inssqs.go (hunk 4 above)
Comment (rafet) [inline inssqs/inssqs.go:132]:
```suggestion
		return ReceiveMessageOutput{}, fmt.Errorf("MaxNumberOfMessages should be less than or equal to 10")
```

**Pull request closed without merging**